Initialized express server, set up routes and pg postgres connection

seanatrinh opened the pull request:
- created basic express server
- created hybrid routes/controller file
- created model file that connects to postgres using pg
- chose Pool, since it supports concurrent requests

seanatrinh committed:
Initialized express server, set up routes pathing, and set up pg connection pool

- created basic express server
- created hybrid routes/controller file
- created model file that connects to postgres using pg
- chose Pool, since it supports concurrent requests

theVikingMan commented on .gitignore:
```diff
@@ -131,3 +131,4 @@ dist
 
 # my additions
 .DS_Store
+.env
```
Believe .env is already incorporated on line 76.

theVikingMan commented on example.env:
```diff
@@ -0,0 +1,2 @@
+# HTTP Port for express server
+PORT="YOUR PORT HERE"
```
Nice job on setting up an example so that someone can look through and see that you put thought into someone else using the code base.

theVikingMan commented on package.json:
```diff
@@ -23,7 +24,9 @@
     "express": "^4.17.3",
-    "mongoose": "^6.3.0"
+    "mongoose": "^6.3.0",
+    "pg": "^8.7.3"
```
You dont have to npm i nodemon here? Unless im not seeing it

theVikingMan commented on server/index.js:
```diff
@@ -0,0 +1,18 @@
+/* eslint-disable no-console */
+require('dotenv').config();
```
In general, i would try to switch uniform adoption of import/export even with node as I found its possible and industry best practice.

theVikingMan commented on server/model.js:
```diff
@@ -0,0 +1,21 @@
+/* eslint-disable no-console */
```
Killer setting up the db file already!

seanatrinh committed:
ESLint + .env changes
- ESLint: Tried to move away from CommonJS modules to allow for import/export syntax in Node files but failed (no major changes here)
- .env: Added Postgres environment variables

seanatrinh committed:
Added express-promise-router
- per node-postgres docs suggestion, added express-promise-router to support use of async/await in express router, which I use in my controller.js file

seanatrinh committed:
Completed first iteration of code for API service, using async/await instead of callbacks
- completed first iteration of API service
- all SQL queries completed (needs optimization, though)
- using async/await instead of callbacks
